src/main/webapp/resources/js/pages/projects/details/ProjectDetails.jsx

```diff
@@ -0,0 +1,142 @@
+import React, { useEffect, useReducer } from "react";
+import {
+  getProjectDetails,
+  updateProjectAttribute,
+} from "../../../apis/projects/projects";
+import { notification, Typography } from "antd";
+import { BasicList } from "../../../components/lists";
+import { formatInternationalizedDateTime } from "../../../utilities/date-utilities";
+import { EditableParagraph } from "../../../components/ant.design";
+import { OntologySelect } from "../../../components/ontology";
+import { TAXONOMY } from "../../../apis/ontology/taxonomy";
+
+const { Paragraph, Title } = Typography;
+
+/**
+ * Reducer for the state of the project details.
+ *
+ * @param {object} state - current state of the project details
+ * @param {object} action - include type and payload
+ * @returns {object}
+ */
+function reducer(state, action) {
+  switch (action.type) {
+    case "LOADED":
+      return { ...state, ...action.payload, loading: false };
+    case "UPDATE":
+      return { ...state, ...action.payload };
+    default:
+      return { ...state };
+  }
+}
+
+/**
+ * React component for render the basic information about a project.
+ * @returns {*}
+ * @constructor
+ */
+export function ProjectDetails() {
+  const [state, dispatch] = useReducer(reducer, { loading: true });
+
+  /*
+  When this component is rendered, query the api for the specific details
+  about this project.
+   */
+  useEffect(() => {
+    getProjectDetails(window.project.id).then((data) =>
+      dispatch({ type: "LOADED", payload: data })
+    );
+  }, []);
+
+  /**
+   * When a field is updated, submitted it to the server to be saved.
+   * Display a notification on the result of the update, and then update the
+   * UI accordingly.
+   *
+   * @param {string} field to be updated
+   * @param {string} value to be store in the field
+   */
+  const updateField = (field, value) => {
```

Comment:
Probably add a check to make sure a value was actually changed and only make a server call if there was a change detected

Reply:
Updated in c6f2992620113f18985e05705da2089dbff7745a, also added a check to ensure that the project label met the requirements for length.